samcli/lib/telemetry/metric.py

```diff
@@ -124,21 +124,28 @@ def hello_command():
 
     """
 
+    @wraps(func)
     def wrapped(*args, **kwargs):
-        telemetry = Telemetry()
-
         exception = None
         return_value = None
         exit_reason = "success"
         exit_code = 0
 
         duration_fn = _timer()
+
+        ctx = None
+        try:
+            ctx = Context.get_current_context()
```

Comment:
What would be the case that it will have no current context and throw a `RuntimeError`?

Reply:
I am not able to find a scenario where if we run a SAM CLI command (eg. `sam build`) we would not have some sort of click context. The only time I was able to find an occurrence of a `RuntimeError` as a result of calling `Context.get_current_context()` is when using the `EventTracker`. This appears to be an extra safe guard from when this was written.